### PR TITLE
Added Date prop type

### PR DESCRIPTION
**Pull request by tshaddix**
Added prop type of `DATE` which attempts to parse date values and returns a…. date

### CODE REVIEW

File: src/message/types.js [hunk 1]
```diff
@@ -26,18 +26,28 @@ module.exports = {
       return undefined;
     }
   },
+  DATE: function(val) {
+    var ms = Date.parse(val);
+
+    if (Number.isNaN(ms)) {
+      return undefined;
```
Comment (tshaddix) [inline src/message/types.js]: Returning undefined from a parse will result in a parse error in the map logic.

Comment (jamsea) [inline src/message/types.js]: Unit test for this? :D

**Comment by jamsea** [comment]
LGTM

**Comment by tshaddix** [comment]
Thanks @jamsea !

**Pull request merged**